### PR TITLE
Add feature: preserve nested folder structure

### PR DESCRIPTION
**Pull request by happycollision**
> Here is another feature I needed 😉

Why:
Previously, attempting to use nested folders with images would result in
`sharp(...).toFile(...)` failure because sharp expects the folder to
already be in place before creating the file. Now images can be nested
inside the `/static/` folder as deeply as desired. The structure will be
recreated in the `/g/` folder.

How:
**Fix incidental bug**
Unrelated: fixed the return statement of `getOutPath()`

**Add a `mkdirp` function**
Sapper supports Node 8 or higher, and the `recursive` option in
`fs.mkdir` was not added until 10.12.0. So we need our own.

**Change `getFileName` to `getRelativePath`**
Instead of getting the basename of the file (assuming it is in the
static folder), we get the relative path using the static folder as a
starting point.

Using the new `mkdirp` function, we can ensure that the nested structure
inside the `/g/` directory exists before each call to `.toFile()`. This
actually removes the need for the `init()` function as well.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ function insert(content, value, start, end, offset) {
 
 function getOutPath(options, filename) {
   const dir = "./static/" + options.outputDir;
-  return (outPath = path.resolve(dir, filename));
+  return path.resolve(dir, filename);
```
Comment (happycollision) [inline index.js]: This was the unrelated fix

**Comment by matyunya** [comment]
Great job, thanks!
